<기본적인 세팅 및 경진대회 데이터 소개>
start with SF_crime data 

## Jupitor Notebook에서 작업

## 작업 폴더 먼저 설정 
import os
os.getcwd()
## 현재 작업 중인 폴더 보여줌
os.chdir('C:\\Users\\USER\\Documents\\GitHub\\San Francisco Crime Kaggle')
## 폴더 변경
os.getcwd()
>> 'C:\\Users\\USER\\Documents\\GitHub\\San Francisco Crime Kaggle'
---
os.chdir('C:\\Users\\USER\\Documents\\GitHub\\San Francisco Crime Kaggle')
---
import pandas as pd
## Load Dataset (여기서 Toggle Toolbar > Markdown 선택하여 글씨체 조정>

train = pd.read_csv("train.csv")
print(train.shape)
train.head()
>> (878049, 9)


###data collumn 
- descript col : 범죄 카테고리에 대한 부연설명 but train data에만 존재, test data에 없음
      -> 참고용으로 쓰지 머신러닝 성능 향상에는 부족할 듯..
- Pd district : 관할 경찰서
- resolution : 현재 해결 여부, 얘도 test data에는 없음, 참고자료 정도 용도.
- address : 수학적, 개발적 지식 뿐만 아니라 domain knowledge 굉장히 중요해, 미국의 경우 address에서 얻을 수 있는 정보가 많다!
       -> 사전조사가 분석에 도움이 될 수 있음

---
## ㅠ.ㅠ 오류남
<csv 파일불러올때 오류>
data = pd.read_csv("D:\\개개비\\multipleChoiceResponses.csv")
 error message : 
OSError: Initializing from file failed

=> 해결방법
1. engin = 'python'
data = pd.read_csv("D:\\개개비\\multipleChoiceResponses.csv", engine='python')
2. 경로 or 파일명 다 영어로
data = pd.read_csv("D:\\multipleChoiceResponses.csv")
---
### engine을 추가하고도 오류가 남, PermissionError: [Errno 13] Permission denied
오류1 :
파이썬에서 파일을 오픈할 때 생겼다.
해결 : 
해당 파일이 존재하는지, 열려 있는지 확인 후 닫고 실행하니 해결되었다.

오류2 :
파이썬에서 엑셀 파일에 쓰고 저장할 때 생겼다.
해결 : 
해당 파일- 속성 > 보안 > Users의 사용 권한 >모든 권한 허용
해주고 실행하니 해결되었다.

-
